### ISSUE TITLE
Fitness example error

### ISSUE THREAD

**Issue by IvBaranov** [flutter/engine#2012]
On save button pressed:
```
E/chromium: [ERROR:dart_error.cc(16)] Unhandled exception:
                                             type 'fitness.UserDataImpl' is not a subtype of type 'fitness.UserDataImpl' of 'data'.
                                             #0      saveFitnessData (/Users/ivbaranov/Projects/flutter/flutter_engine/examples/fitness/lib/user_data.dart:33)
                                             #1      UserDataImpl.save (/Users/ivbaranov/Projects/flutter/flutter_engine/examples/fitness/./lib/main.dart:61)
                                             #2      FitnessAppState._handleItemCreated.<anonymous closure> (/Users/ivbaranov/Projects/flutter/flutter_engine/examples/fitness/./lib/main.dart:106)
                                             #3      State.setState (packages/flutter/src/widgets/framework.dart:382)
                                             #4      FitnessAppState._handleItemCreated (/Users/ivbaranov/Projects/flutter/flutter_engine/examples/fitness/./lib/main.dart:104)
                                             #5      MealFragmentState._handleSave (/Users/ivbaranov/Projects/flutter/flutter_engine/examples/fitness/lib/meal.dart:57)
                                             #6      _RenderInkSplashes._handleTap (packages/flutter/src/material/ink_well.dart:234)
                                             #7      TapGestureRecognizer._checkUp (packages/flutter/src/gestures/tap.dart:83)
                                             #8      TapGestureRecognizer.handlePrimaryPointer (packages/flutter/src/gestures/tap.dart:42)
                                             #9      PrimaryPointerGestureRecognizer.handleEvent (packages/flutter/src/gestures/recognizer.dart:117)
                                             #10     PointerRouter.route (packages/flutter/src/gestures/pointer_router.dart:45)
                                             #11     FlutterBinding.handleEvent (packages/flutter/src/rendering/binding.dart:258)
                                             #12     FlutterBinding.dispatchEvent (packages/flutter/src/rendering/binding.dart:252)
                                             #13     FlutterBinding._handlePointerInputEvent (packages/flutter/src/rendering/binding.dart:237)
                                             #14     FlutterBinding._handleEvent (packages/flutter/src/rendering/binding.dart:194)
                                             #15     _dispatchEvent (file:///src/flutter/engine/src/out/android_Release/gen/sky/bindings/Customhooks.dart:25)
```

Unfortunately, I could not figure out where exactly to submit new issue.
See also https://github.com/dart-lang/sdk/issues/5871

**Comment by abarth** [comment]
Thanks for the bug report.

**Comment by abarth** [comment]
Fixed.

Note: We've moved the examples from this repository to https://github.com/flutter/flutter .  To see the fixed code, you'll probably want to follow the new "getting started" instructions from http://flutter.io/getting-started/ .  In particular, you'll probably want to `pub global deactivate flutter` and use the new `flutter` command from the `bin` directory of https://github.com/flutter/flutter.

Thanks again for the report.
